libs/xb/xenbus_stubs.c

```diff
@@ -65,7 +65,7 @@ CAMLprim value stub_string_of_header(value tid, value rid, value ty, value len)
 	};
 
 	ret = caml_alloc_string(sizeof(struct xsd_sockmsg));
-	memcpy(String_val(ret), &xsd, sizeof(struct xsd_sockmsg));
+	memcpy((void*) String_val(ret), &xsd, sizeof(struct xsd_sockmsg));
```

Comment:
Bytes_val might be better then. You can't memcpy into a string on purpose.

Reply:
There are functions that directly construct a string though caml_alloc_initialized_string